internal/hash.go

```diff
@@ -21,38 +34,78 @@ var algos = map[string]Hasher{
 
 var RecommendedAlgo = "sha256"
 
+// Hasher defines a function that returns a hash.Hash.
 type Hasher func() hash.Hash
 
+// Hash struct encapsulates a hashing algorithm.
 type Hash struct {
 	algo string
 	hash Hasher
 }
 
-var allAlgos = ""
+// Initialize the available algorithms list.
+var allAlgos string
 
+// init initializes the list of available algorithms.
 func init() {
 	initAlgoList()
 }
 
+// initAlgoList initializes the list of available algorithms and verifies the recommended algorithm.
 func initAlgoList() {
-	algoList := make([]string, 0, len(algos))
+	algoNames := make([]string, 0, len(algos))
```

Comment:
This variable name change may be helpful, but it's not really part of this PR.  Make this PR just be about Artifactory.  If you want to propose additional changes, make those be in separate PRs.